planning/obstacle_avoidance_planner/src/utils/utils.cpp

```diff
@@ -694,13 +700,55 @@ bool isOutsideDrivableAreaFromRectangleFootprint(
     tier4_autoware_utils::calcOffsetPose(traj_point.pose, -base_to_rear, -base_to_left, 0.0)
       .position;
 
-  const bool out_top_left = isOutsideDrivableArea(top_left_pos, left_bound, right_bound);
-  const bool out_top_right = isOutsideDrivableArea(top_right_pos, left_bound, right_bound);
-  const bool out_bottom_left = isOutsideDrivableArea(bottom_left_pos, left_bound, right_bound);
-  const bool out_bottom_right = isOutsideDrivableArea(bottom_right_pos, left_bound, right_bound);
+  if (enable_boost_check) {
+    Point2d footprint_point, left_bound_point, right_bound_point, top_bound_point,
+      bottom_bound_point;
+    Polygon2d footprint_polygon;
+    LineString2d left_bound_line, right_bound_line, top_bound_line, bottom_bound_line;
+
+    footprint_point = {top_left_pos.x, top_left_pos.y};
+    footprint_polygon.outer().push_back(footprint_point);
+    footprint_point = {top_right_pos.x, top_right_pos.y};
+    footprint_polygon.outer().push_back(footprint_point);
+    footprint_point = {bottom_right_pos.x, bottom_right_pos.y};
+    footprint_polygon.outer().push_back(footprint_point);
+    footprint_point = {bottom_left_pos.x, bottom_left_pos.y};
+    footprint_polygon.outer().push_back(footprint_point);
+
+    bg::correct(footprint_polygon);
+
+    for (const auto & p : left_bound) {
+      left_bound_point = {p.x, p.y};
+      left_bound_line.push_back(left_bound_point);
+    }
 
-  if (out_top_left || out_top_right || out_bottom_left || out_bottom_right) {
-    return true;
+    for (const auto & p : right_bound) {
+      right_bound_point = {p.x, p.y};
+      right_bound_line.push_back(right_bound_point);
+    }
+
+    top_bound_point = {left_bound_line.back().x(), right_bound_line.back().y()};
+    top_bound_line.push_back(top_bound_point);
+    bottom_bound_point = {left_bound_line.front().x(), right_bound_line.front().y()};
+    bottom_bound_line.push_back(bottom_bound_point);
```

Comment:
I think this should be changed to the following:
```suggestion
    top_bound_line = {left_bound_line.back(), right_bound_line.back()};
    bottom_bound_line = {left_bound_line.front(), right_bound_line.front()};
```